tests/test_pipelines_common.py

```diff
@@ -12,18 +12,86 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from string import ascii_lowercase
 from typing import List, Optional
 from unittest import mock
 
 from transformers import is_tf_available, is_torch_available, pipeline
 from transformers.pipelines import Pipeline
 from transformers.testing_utils import _run_slow_tests, is_pipeline_test, require_tf, require_torch, slow
-from transformers.tokenization_utils_base import to_py_obj
+from transformers.tokenization_utils_base import TruncationStrategy, to_py_obj
 
 
+if is_torch_available():
+    import torch
+
 VALID_INPUTS = ["A simple string", ["list of strings"]]
 
 
+class DummyTok:
```

Comment:
Why not use an existing tokenizer for this? Tokenizers are heavily tested, while this class looks does a lot (encoding, decoding, padding, truncation, converting to framework, creating attention masks, managing several types of inputs) and looks like it should be tested to ensure it behaves correctly.

Reply:
The goal was to have an internet free tokenizer. (I was coding in the train so I really could not use internet there).

I could most definitely make a tokenizer from an existing class but that would require defining a vocabulary on the fly.

The biggest problem with a real tokenizer is the fact that the model is randomly initialized so guaranteeing the output is harder to do. Setting a seed might work, but I don't like to rely on that for tests as it tends to no work as good. Here I simply discard the actual output of the model and simply rely on the shapes.

I do agree that using a Dummy here might be overkill because we need to re implement a lot of functionality and we're not going to be sure that there's not a change between the two implementations.

Would:
- Using an in code vocabulary
- Setting a manual seed
be better in your opinion ?

Reply:
Hmm, we use internet for the tests so it wouldn't be an issue to have an internet-based tokenizer now that you're not in the train!

However, if you really want to go down the road of internet-free tokenizers, we do build some of them in the tokenization tests. Either by using the SentencePiece models in `tests/fixtures/` (see [here](https://github.com/huggingface/transformers/blob/master/tests/test_tokenization_albert.py#L41) an example for ALBERT's tokenizer), or by using an in-code vocabulary as you mention (see [here](https://github.com/huggingface/transformers/blob/master/tests/test_tokenization_bert.py#L44-L78) an example for BERT's tokenizer).

Regarding the randomly initialized model, you can also use a `tiny` variant of a model, which we create specifically for tests. There are already a bunch on the hub, and they sound less dangerous than setting a seed in a test that could potentially have side-effects on other tests down the road.